src/ws/ws_manager.rs

```diff
@@ -95,21 +109,24 @@ impl WsManager {
         let subscriptions = Arc::new(Mutex::new(subscriptions_map));
         let subscriptions_copy = Arc::clone(&subscriptions);
 
+        let stop_flag1 = Arc::clone(&stop_flag);
         let reader_fut = async move {
             // TODO: reconnect
-            loop {
+            while !stop_flag1.load(Ordering::Relaxed) {
                 let data = reader.next().await;
                 if let Err(err) = WsManager::parse_and_send_data(data, &subscriptions_copy).await {
                     error!("Error processing data received by WS manager reader: {err}");
                 }
             }
+            info!("Stopped reading...");
         };
-        spawn(reader_fut);
+        let reader_handle = spawn(reader_fut);
 
-        {
+        let stop_flag2 = Arc::clone(&stop_flag);
```

Comment:
same as above

Reply:
I've made a change but not sure if I understood what you meant correctly